lib/pacemaker/pcmk_sched_colocation.c

```diff
@@ -95,39 +97,37 @@ cmp_primary_priority(gconstpointer a, gconstpointer b)
         return -1;
     }
 
-    CRM_ASSERT(rsc_constraint1->dependent != NULL);
-    CRM_ASSERT(rsc_constraint1->primary != NULL);
+    CRM_ASSERT((rsc_constraint1->dependent != NULL) && (primary1 != NULL));
```

Comment:
check primary2

Reply:
Coverity failed on this because we dereference `rsc_constraint{1,2}` before NULL-checking

Reply:
oops